src/ui/pages/ConnectionDetails/ConnectionDetails.tsx

```diff
@@ -84,9 +84,15 @@ const ConnectionDetails = () => {
         setNotes(connectionDetails.notes);
       }
     } catch (error) {
-      setCloudError(true);
-      // eslint-disable-next-line no-console
-      console.error(error);
+      if (
+        error instanceof Error &&
+        error.message.includes(Agent.MISSING_DATA_ON_KERIA)
+      ) {
+        setCloudError(true);
+      } else {
+        // eslint-disable-next-line no-console
+        console.error("Unable to get connection details", error);
```

Comment:
That's why I was asking earlier @iFergal does the user really need to know all the details behind the error? I think all they need to see is a message saying that the page can't be loaded and that can catch all errors, then we can throw a specific one in the console if we want to... What do you think?

Reply:
Either the error is something we could expect, like the data being missing in the cloud where we can give the option to delete the connection. OR it's an error we don't expect, that could happen for any reason like a networking issue. In this later case, we don't want to ask the user to delete the connection as data is missing in the cloud. That is misleading. We want to say "oops, something unexpected happened, please try again". So no, we have to specifically catch errors.

Reply:
What I'm saying is IF we hit an **unexpected** error, it's generally better to leave it throw/re-throw. BUT, in the UI it would be better to log it, display an error and then clean up the UI so they can try again and not get stuck on a blank page or something (which is why I'm asking what happens when `setLoading` is called but the data isn't set since we hit an error)

Reply:
Yes it makes sense, should we ask Robert to come up with a different design for the "oops, something unexpected happened, please try again" part so we can work on adding that as well, either here or in another PR?

Reply:
Yeah maybe he can just have a generic rule for what to display and not even need to add designs everywhere. Like we display the generic toast and go back to previous page or something. Up to you when we do it

Reply:
Sounds good, I'll leave this as is for now and open a new ticket so we can discuss this with Robert during next refinement. Thank you.